Improve nest camera failure handling on removal

allenporter opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->
## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->
Add failure handling when stopping a camera stream when a camera entity is
removed by swallowing the exception. A camera stream is exentually expired
anyway, so it is safe to ignore the failure and rely on expiration.

Discovered failures here when exercising https://github.com/home-assistant/core/pull/62976 on a dev build where revoking camera permissions would reload the integration, causing it to fail because
Home Assistant no longer had permission to stop the camera.


## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [X] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [ ] The code change is tested and works locally.
- [X] Local tests pass. **Your PR cannot be merged unless tests pass**
- [X] There is no commented out code in this PR.
- [X] I have followed the [development checklist][dev-checklist]
- [X] The code has been formatted using Black (`black --fast homeassistant tests`)
- [X] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.
- [ ] Untested files have been added to `.coveragerc`.

The integration reached or maintains the following [Integration Quality Scale][quality-scale]:
<!--
  The Integration Quality Scale scores an integration on the code quality
  and user experience. Each level of the quality scale consists of a list
  of requirements. We highly recommend getting your integration scored!
-->

- [ ] No score or internal
- [ ] 🥈 Silver
- [ ] 🥇 Gold
- [X] 🏆 Platinum

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html
[quality-scale]: https://developers.home-assistant.io/docs/en/next/integration_quality_scale_index.html
[docs-repository]: https://github.com/home-assistant/home-assistant.io

allenporter committed:
Improve nest camera when removed

Add failure handling when stopping a camera stream when a camera entity is
removed by swallowing the exception. A camera stream is exentually expired
anyway, so it is safe to ignore the failure and rely on expiration.

Discovered failures here when exercising https://github.com/home-assistant/core/pull/62976 on a dev build where revoking camera permissions would reload the integration, causing it to fail because
Home Assistant no longer had permission to stop the camera.

davet2001 commented on homeassistant/components/nest/camera_sdm.py:
```diff
@@ -191,7 +191,12 @@ async def async_will_remove_from_hass(self) -> None:
-            await self._stream.stop_rtsp_stream()
+            try:
+                await self._stream.stop_rtsp_stream()
+            except GoogleNestException as err:
```
Is there a more specific exception that could be used?

The source doesn't seem to be directly browsable but here are the options in the tar.gz:
```
class GoogleNestException(Exception):
    """Base class for all client exceptions."""


class SubscriberException(GoogleNestException):
    """Raised during problems subscribing to events and updates."""


class ApiException(GoogleNestException):
    """Raised during problems talking to the API."""


class AuthException(GoogleNestException):
    """Raised due to auth problems talking to API or subscriber."""


class ConfigurationException(GoogleNestException):
    """Raised due to misconfiguration problems."""


class DecodeException(GoogleNestException):
    """Raised when failing to decode a token."""
```

allenporter replied:
The use of `GoogleNestException` in this file should only apply to `ApiException` and `AuthException`.  I think it could make sense to make `AuthException` derive from `ApiException` and catch that here in this file instead, rather than catching both everywhere here. Let me go do some upstream work, then come back to this.

allenporter replied:
Sent https://github.com/home-assistant/core/pull/63224 with some exception handling updates

allenporter replied:
Updated to use a more narrow exception.

davet2001 reviewed (commented):
Looks good to me, one possible improvement.

allenporter committed:
Use more narrow exception for stop

Use more narrow exception following https://github.com/home-assistant/core/pull/63225

davet2001 reviewed (approved):
Looks good to me, but the test should have a second review by someone with more experience than me!
